tensorflow/python/framework/func_graph.py

```diff
@@ -187,7 +187,7 @@ def __init__(self, name, collections=None, capture_by_value=None):
     self.inputs = []
     self.outputs = []
     self.control_outputs = []
-    self.control_captures = set()
+    self.control_captures = object_identity.ObjectIdentitySet()
```

Comment:
This change is necessary since otherwise the [unittest would fail](https://source.cloud.google.com/results/invocations/62ad6d14-53b0-469a-b8e7-5113f5707ca5/targets) with `TypeError: Variable is unhashable. Instead, use tensor.ref() as the key.`.
See https://github.com/tensorflow/tensorflow/pull/40564#issuecomment-647605662